pkg/operator/boundsatokensignercontroller/controller.go

```diff
@@ -0,0 +1,407 @@
+	"k8s.io/client-go/util/keyutil"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/operatorclient"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+const (
+	workQueueKey = "key"
+
+	operatorNamespace = operatorclient.OperatorNamespace
+	targetNamespace   = operatorclient.TargetNamespace
+
+	keySize                  = 2048
+	NextSigningKeySecretName = "next-bound-service-account-signing-key"
+	SigningKeySecretName     = "bound-service-account-signing-key"
+	PrivateKeyKey            = "service-account.key"
+	PublicKeyKey             = "service-account.pub"
+
+	PublicKeyConfigMapName = "bound-sa-token-signing-certs"
+)
+
+// BoundSATokenSignerController manages the keypair used to sign bound
+// tokens and the key bundle used to verify them.
+type BoundSATokenSignerController struct {
+	operatorClient  v1helpers.StaticPodOperatorClient
+	secretClient    corev1client.SecretsGetter
+	configMapClient corev1client.ConfigMapsGetter
+	eventRecorder   events.Recorder
+
+	cachesSynced []cache.InformerSynced
+
+	// queue only ever has one item, but it has nice error handling backoff/retry semantics
+	queue workqueue.RateLimitingInterface
+}
+
+func NewBoundSATokenSignerController(
+	operatorClient v1helpers.StaticPodOperatorClient,
+	kubeInformersForNamespaces v1helpers.KubeInformersForNamespaces,
+	kubeClient kubernetes.Interface,
+	eventRecorder events.Recorder,
+) *BoundSATokenSignerController {
+
+	ret := &BoundSATokenSignerController{
+		operatorClient:  operatorClient,
+		secretClient:    v1helpers.CachedSecretGetter(kubeClient.CoreV1(), kubeInformersForNamespaces),
+		configMapClient: v1helpers.CachedConfigMapGetter(kubeClient.CoreV1(), kubeInformersForNamespaces),
+		eventRecorder:   eventRecorder.WithComponentSuffix("bound-sa-token-signer-controller"),
+
+		cachesSynced: []cache.InformerSynced{
+			kubeInformersForNamespaces.InformersFor(operatorNamespace).Core().V1().Secrets().Informer().HasSynced,
+			kubeInformersForNamespaces.InformersFor(targetNamespace).Core().V1().Secrets().Informer().HasSynced,
+			kubeInformersForNamespaces.InformersFor(targetNamespace).Core().V1().ConfigMaps().Informer().HasSynced,
+			operatorClient.Informer().HasSynced,
+		},
+
+		queue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "BoundSATokenSignerController"),
+	}
+
+	kubeInformersForNamespaces.InformersFor(operatorNamespace).Core().V1().Secrets().Informer().AddEventHandler(ret.eventHandler())
+	kubeInformersForNamespaces.InformersFor(targetNamespace).Core().V1().Secrets().Informer().AddEventHandler(ret.eventHandler())
+	kubeInformersForNamespaces.InformersFor(targetNamespace).Core().V1().ConfigMaps().Informer().AddEventHandler(ret.eventHandler())
+
+	return ret
+}
+
+func (c *BoundSATokenSignerController) sync() bool {
```

Comment:
the inverse is more standard: true = ok

Reply:
Done. All the more important given that the call site was assuming true = ok too.